src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -87,28 +87,29 @@ public async Task<IActionResult> List(ListContentsViewModel model, PagerParamete
             var pager = new Pager(pagerParameters, siteSettings.PageSize);
 
             // This is used by the AdminMenus so needs to be passed into the options.
-            if (!string.IsNullOrEmpty(contentTypeId))
+            if (!String.IsNullOrEmpty(contentTypeId))
             {
                 model.Options.SelectedContentType = contentTypeId;
             }
 
             // Populate the creatable types.
-            if (!string.IsNullOrEmpty(model.Options.SelectedContentType))
+            if (!String.IsNullOrEmpty(model.Options.SelectedContentType))
             {
                 var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(model.Options.SelectedContentType);
                 if (contentTypeDefinition == null)
                 {
                     return NotFound();
                 }
 
+                var creatableList = new List<SelectListItem>();
+
                 // Allows non creatable types to be created by another admin page.
-                if (model.Options.CanCreateSelectedContentType)
+                if (contentTypeDefinition.GetSettings<ContentTypeSettings>().Creatable || model.Options.CanCreateSelectedContentType)
                 {
-                    model.Options.CreatableTypes = new List<SelectListItem>
-                    {
-                        new SelectListItem(new LocalizedString(contentTypeDefinition.DisplayName, contentTypeDefinition.DisplayName).Value, contentTypeDefinition.Name)
-                    };
+                    creatableList.Add(new SelectListItem(new LocalizedString(contentTypeDefinition.DisplayName, contentTypeDefinition.DisplayName).Value, contentTypeDefinition.Name));
                 }
+
+                model.Options.CreatableTypes = creatableList;
```

Comment:
Idem as below if neither `contentTypeId` nor `SelectedContentType` are provided

So that `model.Options.CreatableTypes` is never null, even there is a `SelectedContentType` that can't be created.